api/docs/http_catalog.md

```diff
@@ -39,11 +39,15 @@ doubly-escaped (it should just be singly-escaped).  To keep backward compatibili
 escaping is still used/supported.
 - `transaction_uuid`: a transaction uuid identifying the entire transaction (shows up in the report as well)
 
+An optional parameter can be provided to the POST or GET to notify a node classifier that the client requested a specific
+environment, which might differ from what the client believes is its current environment.
+- `configured_environment`: the environment configured on the client
+
 ### Example Response
 
 #### Catalog found
 
-    POST /puppet/v3/catalog/elmo.mydomain.com?environment=env
+    POST /puppet/v3/catalog/elmo.mydomain.com?environment=env&configured_environment=env
```

Comment:
The request line isn't quite right. Since it's a POST, the parameters appear in the body of the request, not on the request line:

```
POST /puppet/v3/catalog/arcturus.corp.puppetlabs.net HTTP/1.1
Accept: pson, msgpack, yaml, dot, binary
X-Puppet-Version: 4.2.3
Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3
User-Agent: Ruby
Host: arcturus:8140
Content-Length: 8743
Content-Type: application/x-www-form-urlencoded

environment=production&facts_format=pson&facts=...&transaction_uuid=c09942fd-2c91-4f5b-8602-8f7291911045&fail_on_404=true"
```

I also noticed the agent sends `fail_on_404=true`. I don't know that the server ever uses this parameter (or if it just happens to be a Puppet::Indirector::Request parameter so that it can raise if the server returns 404).